src/conv/k-g/k_parser.cpp

```diff
@@ -477,6 +759,89 @@ bool parse_k
 			}
 			break;
 		    }
+		    case KState::Element_Beam: {
+			if (elementBeamOptions.size() == 0 || elementOptionsCounter == 0) {
+			    if (tokens.size() < 10) {
+				std::cout << "Too short ELEMENT_BEAM in k-file" << fileName << "\n";
+				break;
+			    }
+			    int eid = stoi(tokens[0]);
+
+			    if (data.elements.find(eid) != data.elements.end()) {
+				std::cout << "Duplicat Element ID" << eid << " in k-file" << fileName << "\n";
+				break;
+			    }
+
+			    KElement element;
+
+			    for (int i_n = 0; i_n < 5; ++i_n) {
+				element.nodes.push_back(stoi(tokens[i_n + FirstNode]));
+			    }
+
+			    data.elements[eid] = element;
+
+			    int pid = stoi(tokens[1]);
+			    data.parts[pid].elements.insert(eid);
+			    break;
+			}
+			else if ((elementBeamOptions.size()) > 0 && (elementOptionsCounter < elementBeamOptions.size())) {
+			    Element_Beam_Options currentOption = elementBeamOptions[elementOptionsCounter];
+			    switch (currentOption)
+			    {
+			    case Element_Beam_Options::Thickness: {
+				//handle the thickness line
+				elementOptionsCounter++;

```

Comment:
I prefer `++object` whenever possible:
- `++object` returns the value after `object` was incremented, i.e. a reference to `object`
- `object++` returns the value of `object` before it was incremented; to do this, a copy of `object` has to be created to hold this value, which can be expensive
This doesn't matter in your case here case, but it isn't always obvious, if the extra costs of `object++` are high.

Reply:
this one I didn't know about. thank you for the advice :)